pool/entities.go

```diff
@@ -0,0 +1,170 @@
+package pool
+
+type ID string
+
+type Name string
+
+// I don't know i have to bring Peer Available storage and other metric inside Peer
+type Peer string
+
+type ReplicationFactor int8
+
+// Requirement of joining the pool
+type Requirement struct {
+	Latency   string `json:"latency"`
+	Bandwidth string `json:"bandwidth"`
+	ReplicationFactor ReplicationFactor `json:"rep_factor"`
```

Comment:
Replication factor is not a requirement in joining a pool. it is set by the uploader for each file they want to upload